filebeat/harvester/log.go

```diff
@@ -257,9 +258,16 @@ func (h *Harvester) close() {
 
 	// Make sure file is closed as soon as harvester exits
 	// If file was never opened, it can't be closed
-	if h.file != nil {
+	if h.fileReader != nil || h.file != nil {
+
+		// In case file exists but file reader was never created
+		// Using h.fileReader.Close ensures file handler is only closed once
+		if h.fileReader != nil {
+			h.fileReader.Close()
+		} else {
+			h.file.Close()
+		}
```

Comment:
The same go routine that calls close is also setting the fileReader, so there should not be any race condition.